File: Java_Generated/Java/untitled/src/category_theory/CategoryArrow.java
package category_theory;

import category_theory.interfaces.ICategoryArrow;
import category_theory.interfaces.ICategoryObject;
import diagram.interfaces.IDesktop;
import error_handler.interfaces.ICheck;
import error_handler.interfaces.IErrorHandler;

public class CategoryArrow implements ICategoryArrow {

    protected  ICategoryObject source;

    protected  ICategoryObject target;

    protected String name;

    protected  IDesktop desktop;
    protected IErrorHandler errorHandler;


    protected ICheck checker;

    public CategoryArrow(String name, IDesktop desktop)
    {
        this.name = name;
        this.desktop = desktop;
        if (desktop != null)
        {
            desktop.addCategoryArrow(this);
            checker = desktop.getCheck();
            errorHandler = desktop.getErrorHandler();
        }

    }

    protected boolean check(Object o)
    {
        return checker.check(o);
    }

    protected void show(String message)
    {
        errorHandler.show(message);
    }

    protected void handle(Throwable throwable)
    {
        errorHandler.handle(throwable);
    }


    /**
     * @return
     */
    @Override
    public ICategoryObject getSource() {
        return source;
    }

    /**
     * @return
     */
    @Override
    public ICategoryObject getTarget() {
        return target;
    }

    /**
     * @param source
     */
    @Override
    public void setSource(ICategoryObject source) {
        this.source = source;
    }

    /**
     * @param target
     */
    @Override
    public void setTarget(ICategoryObject target) {
        this.target = target;
    }

    /**
     * @return
     */
    @Override
    public String getArrowName() {
        return name;
    }

    /**
     * @return
     */
    @Override
    public IDesktop getDesktop() {
        return desktop;
    }
}

File: src/Python/category_arrow.py
from typing import Any
from category_object import CategoryObject
from desktop import Desktop

class CategoryArrow:
    def __init__(self, name: str, desktop: Desktop):
        self._name = name
        self._desktop = desktop
        self._source : CategoryObject = None
        self._target : CategoryObject = None
        if self._desktop is not None:
            self._desktop += self
            # TODO add checker and error handler

        pass

    @property
    def name(self) -> str:
        return self._name
    
    @property
    def source(self) -> CategoryObject:
        return self._source
    
    @source.setter
    def source(self, source: CategoryObject):
        self._source = source

    @property
    def target(self) -> CategoryObject:
        return self._source
    
    @target.setter
    def source(self, target: CategoryObject):
        self._target = target

    
    def _check(self, obj: Any) -> bool:
        raise NotImplemented()
    
    def _show(self, message: str) -> None:
        raise NotImplemented()
    
    def _handle(self, exception: Exception) -> None:
        raise NotImplemented
